include/AbcSmc/AbcSmc.h

```diff
@@ -4,298 +4,110 @@
-        void set_posterior_database_filename( std::string name ) { _posterior_database_filename = name; }
-        void set_retain_posterior_rank( std::string retain_rank ) { _retain_posterior_rank = (retain_rank == "true"); }
-        void write_particle_file( const int t );
-        void write_predictive_prior_file( const int t );
-        Metric* add_next_metric(std::string name, std::string short_name, NumericType ntype, double obs_val) {
-            Metric* m = new Metric(name, short_name, ntype, obs_val);
-            _model_mets.push_back(new Metric(name, short_name, ntype, obs_val));
+        void set_retain_posterior_rank( const bool retain_rank ) { _retain_posterior_rank = retain_rank; }
+
+        // TODO: Metric and Parameter management should be private?
+        // if the moving to the "Configuration" class approach, could still have a manual configuration
+        // class / methods on that building class, and not expose this on AbcSmc
+        const ABC::Metric * const add_next_metric(const ABC::Metric * const m) {
+            _model_mets.push_back(m);
             _met_vals.resize(_model_mets.size());
-            _met_vals[_model_mets.size()-1] = obs_val;
+            _met_vals[_model_mets.size()-1] = m->get_obs_val();
             return m;
         }
-        Parameter* add_next_parameter(std::string name, std::string short_name, PriorType ptype, NumericType ntype, double val1, double val2, double step,double (*u)(const double), std::pair<double, double> r, std::map<std::string, std::vector<int> > mm) {
-            Parameter* p = new Parameter(name, short_name, ptype, ntype, val1, val2, step, u, r, mm);
+
+        // TODO model_pars => map<string, Parameter*>, check for duplicate names
+        const ABC::Parameter * const add_next_parameter(const ABC::Parameter * const p) {
             _model_pars.push_back(p);
             return p;
         }
 
-        void set_filtering_type(FilteringType ft) {
-            switch(ft) {
-              case PLS_FILTERING:
-                use_pls_filtering = true; break;
-              case SIMPLE_FILTERING:
-                use_pls_filtering = false; break;
-              default:
-                cerr << "ERROR: Unknown FilteringType: " << ft << endl; exit(-1);
-            }
-        }
+        // should be private?
+        void add_modification_map(
+            const ABC::Parameter * const par,
+            const ABC::ParXform * const xform
+        ) { _par_modification_map[par] = xform; };
+
+        // should be private?
+        void add_par_rescale(
+            const ABC::Parameter * const par,
+            const ABC::ParRescale * const par_rescale
+        ) { _par_rescale_map[par] = par_rescale; }
 
-        FilteringType get_filtering_type() const { return use_pls_filtering ? PLS_FILTERING : SIMPLE_FILTERING; }
+        // should be private?
+        void set_filtering_type(const ABC::FILTER & ft) { _filtering = ft; }
```

Comment:
Remove spaces following & (here and elsewhere)

Reply:
...several of these, but pretty sure all caught.